tests/test_device.py

```diff
@@ -202,6 +202,27 @@ def test_api_protocol_version_is_rotated_with_each_failure(self):
             [call(3.1), call(3.3), call(3.1)]
         )
 
+    def test_api_protocol_version_is_stable_once_successful(self):
+        self.subject._api.set_version.assert_called_once_with(3.3)
+        self.subject._api.set_version.reset_mock()
+
+        self.subject._api.status.side_effect = [
+            {"dps": {"1": False}},
+            Exception("Error"),
+            Exception("Error"),
+            Exception("Error"),
+            Exception("Error"),
+            Exception("Error"),
+            Exception("Error"),
+        ]
+        self.subject.refresh()
+        self.subject.refresh()
+        self.subject.refresh()
+
+        self.subject._api.set_version.assert_has_calls(
+            [call(3.3), call(3.3), call(3.3), call(3.1), call(3.3)]
+        )
```

Comment:
Thanks @make-all, sorry for going back and forward on this. Now that I see this in isolation, I think possibly it's better to avoid the call to `_rotate_api_protocol_version` entirely when we know we've found a working version. Would you mind taking a look at that implementation?